src/dune_rules/foreign_rules.ml

```diff
@@ -18,6 +18,26 @@ let include_dir_flags ~expander ~dir (stubs : Foreign.Stubs.t) =
     let+ lib = Lib.DB.resolve (Scope.libs scope) (loc, lib_name) in
     Lib_info.src_dir (Lib.info lib)
   in
+  let include_path loc include_ =
+    (* Expands the ordered set of paths in the file refered to by [include_].
+       If it expands to a single path, that path is returned.
+       Otherwise a user error is raised. *)
+    let open Action_builder.O in
+    let standard = Action_builder.return [] in
+    let+ paths_str = Expander.expand_and_eval_set expander include_ ~standard in
+    let error msg =
+      User_error.raise ~loc
+        [ Pp.textf "%s" msg
+        ; Pp.textf "Expected target containing singleton (the path to add to include_dirs)."
+        ]
+    in
+    match paths_str with
+    | [ path_str ] -> Path.of_string path_str
+    | [] -> error "`:include` refers to target containing no paths (ie. the empty Ordered Set)."
```

Comment:
Markdown isn't rendered in the terminal, so I suggest we don't include backticks.